doc/source/examples/resolved_cfd-dem/sphere-carreau-with-sharp-interface/sphere-carreau-with-sharp-interface.rst

```diff
@@ -0,0 +1,265 @@
+================================
+Non-Newtonian flow past a sphere
+================================
+
+This example showcases a laminar non-Newtonian flow around a sphere, with an *a priori* Reynolds number Re = 50, using the Carreau rheological model.
+
+Features
+----------------------------------
+- Solvers: ``gls_sharp_navier_stokes_3d`` (with Q1-Q1) 
+- Steady-state problem
+- Non-Newtonian behavior
+- Ramping initial condition
+- Displays the use of non-uniform mesh adaptation 
+
+Location of the example
+------------------------
+
+- Parameter file: ``/examples/resolved_cfd-dem/sphere_carreau_with_sharp_inferface/sphere_carreau_with_sharp_inferface.prm``
+
+
+Description of the case
+-----------------------
+
+In this example, we study the flow around a static sphere using the sharp-interface method to represent the sphere. The geometry of the flow is the following, with a particle of diameter D = 1.0 located at (0,0,0)
+and the flow domain located between (-18,-15,-15) and (42,15,15).
+
+.. image:: images/sharp_carreau_case.png
+    :alt: Simulation schematic
+    :align: center
+
+Parameter file
+-----------------------
+
+Mesh
+~~~~~
+
+The mesh is defined using the following subsection.
+
+.. code-block:: text
+
+	subsection mesh
+	  set type                 = dealii
+	  set grid type            = subdivided_hyper_rectangle
+	  set grid arguments       = 2,1,1 : -18,-15,-15 : 42,15,15 : true
+	  set initial refinement   = 4
+	end
+	
+The dimensions of the used domain are (60x30x30), and the ``subdivided_hyper_rectangle`` is initially divided in (2x1x1) cells, the cells are therefore cubic and of initial size (30x30x30). Using an ``initial refinement`` of 4, the initial size of the cubic cells is 30/2^4 = 1.875. Since the particle size is small in regards to the mesh size, a refinement zone is generated around the particle to better capture it (see :doc:`../../../parameters/cfd/box_refinement` for more details).
+
+.. code-block:: text
+
+	subsection  box refinement
+	  set initial refinement   = 3
+	  subsection mesh
+	      set type                 = dealii
+	      set grid type            = subdivided_hyper_rectangle
+	      set grid arguments       = 1,1,1: -2,-2,-2 : 6,2,2 : true
+	      set initial refinement   = 0
```

Comment:
Very good point :) the initial refinement of the mesh itself is zero, it should always be as small as possible since when using an box refinement, only the mesh boundaries matter. The time taken to refine the mesh inside the boundaries of the box is O(n_cells_of_mesh*n_cells_in_box_mesh), so it's just much faster to generate the same box when using fewer box mesh cells. Let me know if this still isn't clear. The ``initial refinement = 3`` sets how many times the cells inside the boundaries of the hyper_rectangle are refined.

Reply:
I also had a hard time understanding at first!